mod/search/classes/Elgg/Search/FormatComentEntityHook.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Elgg\Search;
+
+use Elgg\Hook;
+use ElggEntity;
+
+/**
+ * Populates comment's volatile data for search listing rendering
+ *
+ * @elgg_plugin_hook search:format entity
+ */
+class FormatComentEntityHook {
+
+	/**
+	 * Format comment entity in search results
+	 *
+	 * @elgg_plugin_hook search:format entity
+	 *
+	 * @param Hook $hook Hook
+	 * @return ElggEntity
+	 */
+	public function __invoke(Hook $hook) {
+
+		$entity = $hook->getValue();
+
+		if (!elgg_instanceof($entity, 'object', 'comment')) {
+			return;
+		}
+
+		$owner = $entity->getOwnerEntity();
+		$size = $hook->getParam('size', 'small');
+		$icon = elgg_view_entity_icon($owner, $size);
+
+		$container = $entity->getContainerEntity();
+		if (!$container instanceof ElggEntity) {
+			return;
+		}
+
+		$title = $container->getDisplayName();
+
+		if (!$title) {
+			$title = elgg_echo('item:' . $container->getType() . ':' . $container->getSubtype());
+		}
+
+		if (!$title) {

```

Comment:
can you get here, as an `elgg_echo()` of a non-existing language key results in the key being echoed

Reply:
could you clarify?

Reply:
the `if` above this one (line 42) is almost the same, but it checks a different language key. So i think the first `if` (line 42) will make sure you can never get into this one.

Reply:
maybe if the language key exists but it's an empty string...